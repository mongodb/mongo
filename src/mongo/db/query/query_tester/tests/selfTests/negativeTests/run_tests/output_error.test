output_error
negative_test
test.coll as coll

// This test should execute successfully and be numbered 0.
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toString": "$value" } } } ], cursor: {} }

// This test should fail and be numbered 1.
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toDate": "$value" } } } ], cursor: {} }

// This test should execute successfully and be numbered 2.
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toLong": "$_id" } } } ], cursor: {} }
