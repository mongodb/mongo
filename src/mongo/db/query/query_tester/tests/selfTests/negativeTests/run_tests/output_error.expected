output_error
negative_test
test.coll as coll

// This test should execute successfully and be numbered 0.
0
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toString": "$value" } } } ], cursor: {} }
[
{"_id":0,"projected":"a","value":"a"},
{"_id":1,"projected":"b","value":"b"}
]

// This test should fail and be numbered 1.
1
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toDate": "$value" } } } ], cursor: {} }
Failed to execute test number 1. Expected OK command result but got { ok: 0.0, errmsg: "Executor error during aggregate command on namespace: negative_test.coll :: caused by :: an incomplete date/time string has been found, with elements ...", code: 241, codeName: "ConversionFailure" }

// This test should execute successfully and be numbered 2.
2
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toLong": "$_id" } } } ], cursor: {} }
[
{"_id":0,"projected":0,"value":"a"},
{"_id":1,"projected":1,"value":"b"}
]
